java/sage/media/sub/SubtitleHandler.java

```diff
@@ -121,7 +134,11 @@ else if (rv.length() == 0 && willDisplay)
         }
       }
     }
-    //System.out.println("Got Subtitle text=" + rv + " subEntryPos=" + subEntryPos);
+    finally
+    {
+      subtitleLock.readLock().unlock();
+    }
+    System.out.println("Got Subtitle text=" + rv + " subEntryPos=" + subEntryPos);
```

Comment:
You may want to get rid of this debug line...I'd imagine it logs a LOT when you do subtitle playback. You may also want to add some kind of "debug_subs" property which controls logging in the subtitle classes too.